dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/DatasetFieldWalker.java

```diff
@@ -48,22 +48,21 @@
      */
     public static void walk(DatasetField dsf, Listener l) {
         DatasetFieldWalker joe = new DatasetFieldWalker(l);
-        SettingsServiceBean nullServiceBean = null;
-        joe.walk(dsf, nullServiceBean);
+        joe.walk(dsf, true);
     }
 
     /**
      * Convenience method to walk over a list of fields. Traversal
      * is done in display order.
      *
      * @param fields  the fields to go over. Does not have to be sorted.
-     * @param exclude the fields to skip
+     * @param excludeEmailFromExport is email excluded from export
      * @param l       the listener to execute on each field values and structure.
      */
-    public static void walk(List<DatasetField> fields, SettingsServiceBean settingsService, Listener l) {
+    public static void walk(List<DatasetField> fields, Listener l, boolean excludeEmailFromExport) {
```

Comment:
I'm wondering if `excludeEmailFromExport` is the best name for this parameter. This class is unaware of export.
I think that `excludeEmailFieds` would be better name